### ISSUE TITLE
Add unique instruction Id

### ISSUE THREAD

**Issue by flobernd** [zyantific/zydis#104]

**Comment by ZehMatt** [comment]
I had recently a brief discussion with @athre0z about something similar, zasm currently generates the instruction nodes via temporarily encoding and decoding the instruction to obtain the instruction meta information such as operand access and CPU flags, later during serialization it will fully encode the instruction. So one idea is to be able to get the id of the instruction and be able to pass the id later to the encoder back to avoid the lookups for each encode, also an additional API to query the meta data by that id so for zasm I wouldn't need to encode and decode and rather obtain the unique id and store that instead. I think the encoder and decoder can both use something like ZydisInstructionGetMeta(id, res).

**Comment by flobernd** [comment]
@ZehMatt Yes, it will have several positive side effects. There is some (unintentional) preperation for that in this PR: https://github.com/zyantific/zydis/pull/297/files#diff-c323a181d40c83b95974655308d8abeb7cba5a1e16b6c701fd77b53a35e72f55L1258

After the final unique ID implementation, this field will be moved to the `ZydisDecodedInstruction`.
